crates/sync/anvil/src/tracker.rs

```diff
@@ -215,17 +217,29 @@ async fn process(ctx: Ctx, mut block_rcv: mpsc::UnboundedReceiver<Msg>) -> Resul
             }
         }
 
-        for erc721_token_info in ctx
+        for erc721_token in ctx
             .db
-            .get_erc721_missing_metadata(ctx.chain_id)
+            .get_erc721_missing_data(ctx.chain_id)
             .await?
             .into_iter()
         {
-            let token_id = erc721_token_info.token_id;
-            let address = erc721_token_info.contract;
-            let metadata = fetch_erc721_metadata(erc721_token_info, &provider).await;
+            let token_id = erc721_token.token_id;
+            let address = erc721_token.contract;
+            let owner = erc721_token.owner;
+            let erc721_data = fetch_erc721_data(erc721_token, &provider).await;
+
+            dbg!(&erc721_data);
             ctx.db
-                .save_erc721_metadata(address, ctx.chain_id, token_id, metadata)
+                .save_erc721_data(
+                    address,
+                    ctx.chain_id,
+                    token_id,
+                    owner,
+                    erc721_data.name,
+                    erc721_data.symbol,
+                    erc721_data.uri,
+                    erc721_data.metadata,
+                )
```

Comment:
This loop is once again mixing two concepts: the collection, and the individual item

`db.get_erc721_missing_data` only checks for missing collection data. meaning that it will skip items for which the collection exists, but the item metadata doesn't. that means they'll never get updated

These should happen as two independent tasks: update any collections. and updating any items